### PR TITLE
Add overview assembly

### PR DESCRIPTION
**Pull request by scholzj**
This PR adds the overview assembly with following chapters:
* Overview
  * Key Features
  * Supported Configurations
  * Document Conventions

### CODE REVIEW

File: books/assembly-overview.adoc [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+= Overview of AMQ Streams
+
+Red Hat AMQ Streams is a massively-scalable, distributed, and high-performance data streaming platform based on the Apache Zookeeper and Apache Kafka projects.
+It consists from 5 main components:
+
+Zookeeper:: Service for highly reliable distributed coordination.
+Kafka Broker:: Messaging broker responsible for delivering records from producing clients to consuming clients.
+Kafka Connect:: A toolkit for streaming data between Kafka brokers and other systems using _Connector_ plugins.
+Kafka Consumer and Producer APIs:: Java based APIs for producing and consuming messages to / from Kafka brokers.
+Kafka Streams API:: API for writting _stream processor_ applications.
```
Comment (ppatierno) [inline books/assembly-overview.adoc]: typo "writing"

File: books/assembly-overview.adoc [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+
+// Save the context of the assembly that is including this one.
+// This is necessary for including assemblies in assemblies.
+// See also the complementary step on the last line of this file.
+
+[id='overview-{context}']
+= Overview of AMQ Streams
+
+Red Hat AMQ Streams is a massively-scalable, distributed, and high-performance data streaming platform based on the Apache Zookeeper and Apache Kafka projects.
+It consists from 5 main components:
```
Comment (dmuntima) [inline books/assembly-overview.adoc]: It consists from 5 main components:> It consists the following main components:

File: books/assembly-overview.adoc [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+
+Zookeeper:: Service for highly reliable distributed coordination.
+Kafka Broker:: Messaging broker responsible for delivering records from producing clients to consuming clients.
+Kafka Connect:: A toolkit for streaming data between Kafka brokers and other systems using _Connector_ plugins.
+Kafka Consumer and Producer APIs:: Java based APIs for producing and consuming messages to / from Kafka brokers.
+Kafka Streams API:: API for writing _stream processor_ applications.
+
+Kafka Broker is the central point connecting all these components. 
+The broker itself uses Apache Zookeeper for storing configuration data and for cluster coordination (electing leaders, and so on).
```
Comment (dmuntima) [inline books/assembly-overview.adoc]: delete "itself" and  (electing leaders, and so on).

File: books/assembly-overview.adoc [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+
+Zookeeper:: Service for highly reliable distributed coordination.
+Kafka Broker:: Messaging broker responsible for delivering records from producing clients to consuming clients.
+Kafka Connect:: A toolkit for streaming data between Kafka brokers and other systems using _Connector_ plugins.
+Kafka Consumer and Producer APIs:: Java based APIs for producing and consuming messages to / from Kafka brokers.
+Kafka Streams API:: API for writing _stream processor_ applications.
+
+Kafka Broker is the central point connecting all these components. 
+The broker itself uses Apache Zookeeper for storing configuration data and for cluster coordination (electing leaders, and so on). 
+Before installing Apache Kafka, an Apache Zookeeper cluster has to be ready.
```
Comment (dmuntima) [inline books/assembly-overview.adoc]: We are packaging AMQ Streams in a single zip for all the components installation. I think this statement is incorrect.

File: books/ref-overview-document-conventions.adoc [hunk 5]
```diff
@@ -0,0 +1,17 @@
+// Module included in the following assemblies:
+//
+// assembly-overview.adoc
+
+[id='overview-document-conventions-{context}']
+= Document Conventions
```
Comment (dmuntima) [inline books/ref-overview-document-conventions.adoc]: Document Conventions > Document conventions  (change to title case)

File: books/ref-overview-key-features.adoc [hunk 6]
```diff
@@ -0,0 +1,15 @@
+// Module included in the following assemblies:
+//
+// assembly-overview.adoc
+
+[id='overview-key-features-{context}']
+= Key Features
```
Comment (dmuntima) [inline books/ref-overview-key-features.adoc]: Key Features >  Key features (change to title case)

File: books/ref-overview-supported-configurations.adoc [hunk 7]
```diff
@@ -0,0 +1,24 @@
...
+.List of supported Java Virtual Machines
+[width="80%",cols="3,^2",options="header"]
+|=========================================================
+|Java Virtual Machine |Version
+|OpenJDK |1.8
+|OracleJDK |1.8
+|IBM JDK |1.8
+|=========================================================
+
+.List of supported Operating Systems
```
Comment (dmuntima) [inline books/ref-overview-supported-configurations.adoc]: Add Apache Kafka version.

File: books/assembly-overview.adoc (hunk 2 above)
Comment (scholzj) [reply books/assembly-overview.adoc]: _It consists the following main component_ ... doesn't it miss _from_ or _of_?

File: books/assembly-overview.adoc (hunk 4 above)
Comment (scholzj) [reply books/assembly-overview.adoc]: Yeah, I changed it to _Before running Apache Kafka, an Apache Zookeeper cluster has to be ready._

File: books/ref-overview-supported-configurations.adoc (hunk 7 above)
Comment (scholzj) [reply books/ref-overview-supported-configurations.adoc]: Could you clarify what do you mean with that? AMQ Streams == Kafka. We do not support any Kafka versions.

Comment (dmuntima) [reply books/ref-overview-supported-configurations.adoc]: We added Kafka version number in the OCP guide here:https://access.redhat.com/documentation/en-us/red_hat_amq_streams/1.0-beta/html-single/using_amq_streams/#getting-started-str 

I was thinking on the same lines and wanted to add the Kafka/upstream version number.

Comment (scholzj) [reply books/ref-overview-supported-configurations.adoc]: Yeah ... I think it makes more sense there as Strimzi is much more than just Kafka. But the standalone version is just Apache Kafka. And I think that for example for AMQ Broker the docs also don't say that it supports Apache ActiveMQ Artemis 2.3.0 which would be the same as here.

**Pull request merged**